File: Official Windows Platform Sample/Windows 8.1 Store app samples/99866-Windows 8.1 Store app samples/XAML requested theme sample/C#/UserSettings.cs
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

using System.ComponentModel;
using Windows.UI.Xaml;

namespace RequestedTheme
{
    // This class is used in Scenarios 3 and 4.
    public class UserSettings : INotifyPropertyChanged
    {
        private ElementTheme selectedTheme = ElementTheme.Dark;
        public ElementTheme SelectedTheme
        {
            get
            {
                return selectedTheme;
            }
            set
            {
                selectedTheme = value;
                OnPropertyChanged(new PropertyChangedEventArgs("SelectedTheme"));
            }
        }
        
        public event PropertyChangedEventHandler PropertyChanged;

        public void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, e);
            }
        }

    }
}

File: Official Windows Platform Sample/Windows 8.1 Store app samples/99866-Windows 8.1 Store app samples/XAML requested theme sample/C++/UserSettings.cpp
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//
// Copyright (c) Microsoft Corporation. All rights reserved

//
// UserSettings.cpp
// Implementation of the UserSettings class
// This class is used in Scenarios 3 and 4.
//

#include "pch.h"
#include "UserSettings.h"

using namespace SDKSample::RequestedThemeCPP;
using namespace Platform;
using namespace Windows::UI;
using namespace Windows::UI::Xaml::Data;
using namespace Windows::UI::Xaml::Media;

UserSettings::UserSettings()
{
	_selectedTheme = Windows::UI::Xaml::ElementTheme::Dark;
}

void UserSettings::SelectedTheme::set(Windows::UI::Xaml::ElementTheme value)
{
    if (_selectedTheme != value)
    {
       _selectedTheme = value;
        OnPropertyChanged("SelectedTheme");
    }
}

Windows::UI::Xaml::ElementTheme UserSettings::SelectedTheme::get()
{
    return _selectedTheme;
}

void UserSettings::OnPropertyChanged(String^ propertyName)
{
    PropertyChanged(this, ref new PropertyChangedEventArgs(propertyName));
}
